src/Engine/Graphics/Nuklear.cpp

```diff
@@ -3295,149 +3526,119 @@ static int lua_unset_hotkeys(lua_State *L) {
 
+    int x = 0, y = 0;
+    if (lua_gettop(L) >= 1) {
+        x = luaL_checkint(L, 1);
+    }
+    if (lua_gettop(L) >= 2) {
+        y = luaL_checkint(L, 2);
+    }
+    nk_scroll* scroll = (nk_scroll*)lua_newuserdata(L, sizeof(nk_scroll));
+    scroll->x = x;
+    scroll->y = y;
+    luaL_getmetatable(L, "nk_scroll_mt");
+    lua_setmetatable(L, -2);
     return 1;
 }
 
-static int lua_party_give(lua_State *L) {
-    lua_check_ret(lua_check_args(L, lua_gettop(L) == 3));
-
-    const char *name = luaL_checkstring(L, 1);
-    int value = luaL_checknumber(L, 2);
-
-    if (!strcmp(name, "food"))
-        pParty->GiveFood(value);
-    else if (!strcmp(name, "gold"))
-        pParty->AddGold(value);
-    else
-        return luaL_argerror(L, 2, lua_pushfstring(L, "name '%s' is unknown", name));
-
-    return 0;
-}
-
-static int lua_party_set(lua_State *L) {
-    lua_check_ret(lua_check_args(L, lua_gettop(L) == 3));
-
-    const char *name = luaL_checkstring(L, 1);
-    int value = luaL_checknumber(L, 2);
-
-    if (!strcmp(name, "food"))
-        pParty->SetFood(value);
-    else if (!strcmp(name, "gold"))
-        pParty->SetGold(value);
-    else
-        return luaL_argerror(L, 2, lua_pushfstring(L, "name '%s' is unknown", name));
+static int lua_nk_scroll_set(lua_State *L) {
+    lua_check_ret(lua_check_args_count(L, lua_gettop(L) >= 3));
 
+    nk_scroll* scroll = lua_nk_check_scroll(L, 1);
+    scroll->x = luaL_checkint(L, 2);
+    scroll->y = luaL_checkint(L, 3);
     return 0;
 }
 
-static int lua_party_take(lua_State *L) {
-    lua_check_ret(lua_check_args(L, lua_gettop(L) == 3));
+static int lua_dev_config_set(lua_State *L) {
+    lua_check_ret(lua_check_args_count(L, lua_gettop(L) >= 2 && lua_gettop(L) <= 3));
 
-    const char *name = luaL_checkstring(L, 1);
-    int value = luaL_checknumber(L, 2);
-
-    if (!strcmp(name, "food"))
-        pParty->TakeFood(value);
-    else if (!strcmp(name, "gold"))
-        pParty->TakeGold(value);
-    else
-        return luaL_argerror(L, 2, lua_pushfstring(L, "name '%s' is unknown", name));
+    int valueIndex = 3;
+    const char *configName{};
+    AnyConfigEntry* configEntry{};
+    if (lua_gettop(L) > 2) {
+        const char *sectionName = luaL_checkstring(L, 1);
+        configName = luaL_checkstring(L, 2);
+        ConfigSection* section = engine->config->section(sectionName);
+        if (section != nullptr) {
+            configEntry = section->entry(configName);
+        } else {
+            return luaL_argerror(L, 2, lua_pushfstring(L, "invalid section name: '%s'", sectionName));
+        }
+    } else {
+        configName = luaL_checkstring(L, 1);
+        for (auto&& section : engine->config->sections()) {
+            configEntry = section->entry(configName);
+            if (configEntry != nullptr) {
+                break;
+            }
+        }
+        valueIndex = 2;
+    }
+
+    if (configEntry != nullptr) {
+        if (configEntry->type() == typeid(bool)) {
```

Comment:
Just use `configEntry->setString(lua_tostring(L, valueIndex))`?

Reply:
In order to call the correct lua_totype conversion I need to check the types of each config. I could give to the lua script calling the function this responsibility but it didn't seem correct.

Reply:
Can't we use `lua_tolstring`?

Reply:
I was under an impression that `lua_tostring` converts the value on the lua stack in place into a string, so it should just work. No?

Reply:
Ah, nvm! I confused this method with the opposite one. I didn't know configEntry had a setString that does the deserialization for us. That's way better of course. I'll change it.